raiden/tests/unit/transfer/mediated_transfer/test_targetstate.py

```diff
@@ -273,16 +283,31 @@ def test_handle_onchain_secretreveal():
     assert factories.UNIT_SECRETHASH in channel_state.partner_state.secrethashes_to_unlockedlocks
     assert factories.UNIT_SECRETHASH not in channel_state.partner_state.secrethashes_to_lockedlocks
 
+    # Make sure that an emptyhash on chain reveal is rejected.
     block_number_prior_the_expiration = expiration - 2
+    onchain_reveal = ContractReceiveSecretReveal(
+        transaction_hash=factories.make_address(),
+        secret_registry_address=factories.make_address(),
+        secrethash=EMPTY_HASH_KECCAK,
+        secret=EMPTY_HASH,
+        block_number=block_number_prior_the_expiration,
```

Comment:
Does this make sense? the reason we're doing this is because the contract rejects zero hash... do you think we can just trust that the contract will never accept a zero-hash?

Reply:
> do you think we can just trust that the contract will never accept a zero-hash

The contract does not accept a zero hash at the moment so yes.

Reply:
Yup, that's what we agreed on... initially i to remove the checck... but i think  it to make sure that the client will behave properly if the contracts do accept this hash value. So let's keep it for now.